### ISSUE TITLE
just a comment

### ISSUE THREAD

**Issue by iandoug** [ellisgl/keyboard-schema#2]
Hi

I was also thinking we need something like this.

Might want to have a better data structure than Typescript though ... something language-agnostic, for wider compatibility.

I did think of using YAML but worried about editors accidentally deleting spaces and messing things up .... :-)

**Comment by ellisgl** [comment]
Yeah I started off using JSON Schema, and switched to TypeScript not thinking of being interchangable. I think I could go back to JSON Schema, since pretty much every modern language supports JSON. Also I want to make sure it has fairly strict typing.

**Comment by ellisgl** [comment]
@iandoug OK - I've swapped back using JSON schema. Any help improving these would be great. I'm trying to keeps these boiled down to what is needed, and not going overboard like I did the first time where I was defining way more than was is actually need to design a keyboard.

**Comment by iandoug** [comment]
At the moment we have multiple formats for defining a keyboard, whether physical or logical. eg
Microsoft (klc?)
Apple (Ukulele?)
Linux/etc (xkb)

Keyboard Layout Editor (2 versions)
Keyboard Layout Analyzer
QMK
assorted "shorthand" versions posted on forums
probably some others too

So I was looking for a Babelfish to translate between them...

I don't know what level of detail we need to get to .. if too much work to enter, then most users won't bother. 

Might be an idea to have a basic flexible scheme first, and then see what NEEDS to be added?

BTW KLE has a lot of switch data that you can borrow.

**Comment by ellisgl** [comment]
Yeah, there are probably more things I could do do with switch definitions, but I was getting crazy with like clicky, linear, force Newton's, stuff like that. 

I was thing mostly physical definitions, that way things could be used for creating blender, cad, schematic/PCB files. But something that can help with QMK firmware would be great too.

I have a PR for increasing the number of decimal places that could used in KLE, but it's been a couple years since a PR has been pulled in and I don't think there will be an update anytime soon and merge conflicts are going to happen. So I wanted to create a set of definitions so people could make tools around them, and there would be sort of "standard". 

With my time being limited and just my luck in "marketing", I don't have high expectations that many people will adopt what I've done, but will continue on because I want to at least get something out there.
